### ISSUE TITLE
Autocomplete/IntelliSense works only under very specific circumstances

### ISSUE THREAD

**Issue by lukebarton** [flowtype/flow-language-server#74]
Originally posted in [flow-for-vscode](https://github.com/flowtype/flow-for-vscode/issues/227) but I imagine the issue actually lies with the language server since the behaviour is **identical in Atom**, so raising here retrospectively.

![flow-autocomplete-issue](https://user-images.githubusercontent.com/98427/38336986-a8735408-385b-11e8-867e-d7d26346f3ff.gif)

Maybe I'm doing something completely wrong but I've been trying to get flow working nicely in VSCode for several hours. I've concluded that IntelliSense with flow has some real issues and recorded this gif showing two ways I believe it's broken:

1. Chaining causes IntelliSense to stop working.
2. Trying to use it preceding an anonymous function without a semicolon somewhere between causes IntelliSense to stop working.

These issues may be related. Needless to say, it's making Flow pretty difficult to work with.

flow 0.69.0
flow-for-vscode 0.8.0

**Comment by namuol** [comment]
@lukebarton Are you using the `flow.useLSP: true` in your VSCode config?

https://github.com/flowtype/flow-for-vscode/blob/438cad0395be64cb4cf337648d595742354fd5b7/package.json#L55-L58

**Comment by lukebarton** [comment]
Yes, but the option makes no difference. I think I jumped the gun a bit - I
didn't realise that the language server wasn't the default implementation
yet. I think the issue lies somewhere more core to flow than just the
language server or the vscode implementation. It seems what flow itself is
returning isn't as expected.


On 28 Apr 2018 at 23:07, <Lou Acresti <notifications@github.com>> wrote:

@lukebarton <https://github.com/lukebarton> Are you using the flow.useLSP:
true in your VSCode config?

https://github.com/flowtype/flow-for-vscode/blob/438cad0395be64cb4cf337648d595742354fd5b7/package.json#L55-L58

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/flowtype/flow-language-server/issues/74#issuecomment-385208671>,
or mute the thread

**Comment by namuol** [comment]
@lukebarton Hmm -- technically `flow-language-server` _does_ implement autocompletion, as far as I can tell:

https://github.com/flowtype/flow-language-server/blob/daa83c3bc5d6323f0e1ce17bfd38c04ad43f52b9/packages/flow-language-server/src/Completion.js#L43-L46

Try disabling `@builtin typescript` in VSCode _and_ enabling `useLSP`, [like so](https://cl.ly/1p1h3r2V2X1R).

After doing that, give it another try -- sometimes the issue is just that Flow itself doesn't know enough about the type 🤞 Good luck!

**Comment by lukebarton** [comment]
TypeScript builtin is disabled. I tried both LSP on and off. The behaviour
is the same in Atom as with VS Code.

On 1 May 2018 at 00:16, <Lou Acresti <notifications@github.com>> wrote:

@lukebarton <https://github.com/lukebarton> Hmm -- technically
flow-language-server *does* implement autocompletion, as far as I can tell:

https://github.com/flowtype/flow-language-server/blob/daa83c3bc5d6323f0e1ce17bfd38c04ad43f52b9/packages/flow-language-server/src/Completion.js#L43-L46

Try disabling @builtin typescript in VSCode *and* enabling useLSP, like so
<https://cl.ly/1p1h3r2V2X1R>.

After doing that, give it another try -- sometimes the issue is just that
Flow itself doesn't know enough about the type 🤞 Good luck!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/flowtype/flow-language-server/issues/74#issuecomment-385556438>,
or mute the thread

**Comment by namuol** [comment]
@lukebarton If you're using LSP and autocomplete isn't working, my guess would be that Flow actually doesn't know enough about the type of the symbol you're interacting with to provide you with completion features -- try hovering over the symbol to see what Flow reports its type as.

Otherwise, it may indeed be a bug with the LSP implementation -- without more detailed information it'll be hard for us to know; @wbinnssmith any recommendations on how to provide debugging information to bug reports like this one?

**Comment by namuol** [comment]
> I posted a gif in the original post showing what happens.

Yeah I just wasn't sure if that gif was still the _exact_ behavior, as you mentioned you may have posted that while _not_ using `useLSP`:

> I think I jumped the gun a bit - I didn't realise that the language server wasn't the default implementation yet.

Didn't mean to belittle your experience! The behavior in the gif is really confusing/frustrating, for sure. I'm just trying to clarify so we can track down the issue with precision -- Thanks! 🙇

**Comment by lukebarton** [comment]
So I tried with LSP on and off, nothing changed.

Sorry for replying with a numbered list. It's been a long day and I didn't want to write the fluffy nice stuff you usually have to - there's no beef/ill feeling 👍

**Comment by namuol** [comment]
> Sorry for replying with a numbered list.

No worries 😂  Thanks for replying 👍


> So I tried with LSP on and off, no difference. On/Off/VS Code/Atom all the same.

> It's probably a flow core thing.

_Maybe_, but the behavior in that gif still looks really odd. When you find the time, would you mind recording the same sequences of behaviors with LSP on in VSCode? It may be some combination of both Flow's core "type of symbol" reporting that can be addressed in the LSP integration 🤔
